src/paho/mqtt/client.py

```diff
@@ -3328,16 +3338,44 @@ def _handle_publish(self):
             return MQTT_ERR_SUCCESS
         elif message.qos == 1:
             self._handle_on_message(message)
-            return self._send_puback(message.mid)
+            if self._manual_ack:
+                return MQTT_ERR_SUCCESS
+            else:
+                return self._send_puback(message.mid)
         elif message.qos == 2:
+
             rc = self._send_pubrec(message.mid)
+
             message.state = mqtt_ms_wait_for_pubrel
             with self._in_message_mutex:
                 self._in_messages[message.mid] = message
+
             return rc
         else:
             return MQTT_ERR_PROTOCOL
 
+    def ack( self, mid, qos ):
+        """
+           send an acknowledgement for a given message id. (stored in message.mid )
+           only useful in QoS=1 and auto_ack=False
+        """
+        if self._manual_ack :
+            if qos == 1:
+                return self._send_puback(mid)
+            elif qos == 2:
+                return self._send_pubcomp(mid)
+
+        return MQTT_ERR_SUCCESS
+
+    def manual_ack( self, on=False ):
```

Comment:
```suggestion
    def manual_ack_set(self, on):
```

I think naming it `thing_set` is better. It sets an option. `set_manual_ack` sound more natural, but all other setter use `thing_set` (`tls_set`, `reconnect_delay_set`...)

On the non-default value I'm a bit mitigated. For consistency we should have a default. But it feels really odd. You call `manual_ack_set()` and it get disabled ? I think `manual_ack_set` has one flag which should be required, even if that not very consistent with other setter.